clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockOutputStream.java

```diff
@@ -259,15 +258,23 @@ private void ensureNotFinished() {
 
     @Override
     public void close() throws IOException {
-        if (!finished) {
-            // basically flush the buffer writing the last block
-            writeBlock();
-            // write the end block and finish the stream
-            writeEndMark();
-        }
-        if (out != null) {
-            out.close();
-            out = null;
+        try {
+            if (!finished) {
+                // basically flush the buffer writing the last block
+                writeBlock();
+                // write the end block
+                writeEndMark();
+            }
+        } finally {
+            try {
+                if (out != null)
+                    out.close();
+            } finally {
+                out = null;
```

Comment:
We should probably inherit from `OutputStream` and control the lifetime of this variable ourselves. `FilterOutputStream` expects `out` to be non null. We override all of its methods, but then there's no real benefit in inheriting from that. Thoughts?

Reply:
Makes sense, updated.